plumbing/format/diff/patch.go

```diff
@@ -0,0 +1,72 @@
+package diff
+
+import (
+	"io"
+
+	"gopkg.in/src-d/go-git.v4/plumbing"
+	"gopkg.in/src-d/go-git.v4/plumbing/filemode"
+)
+
+// Operation defines the operation of a diff item.
+type Operation int
+
+const (
+	// Equal item represents a equals diff.
+	Equal Operation = iota
+	// Add item represents an insert diff.
+	Add
+	// Delete item represents a delete diff.
+	Delete
+)
+
+// Patch represents the necessary steps to transform several files to another.
```

Comment:
English can be improved a little bit by simplifying the statement:

`Patch represents a collection of steps to transform several files`